airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumPropertiesManager.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.debezium.internals;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.db.jdbc.JdbcUtils;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
+import io.airbyte.protocol.models.SyncMode;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import org.codehaus.plexus.util.StringUtils;
+
+public class DebeziumPropertiesManager {
+
+  private final JsonNode config;
+  private final AirbyteFileOffsetBackingStore offsetManager;
+  private final Optional<AirbyteSchemaHistoryStorage> schemaHistoryManager;
+
+  private final Properties properties;
+  private final ConfiguredAirbyteCatalog catalog;
+
+  public DebeziumPropertiesManager(final Properties properties,
+                                   final JsonNode config,
+                                   final ConfiguredAirbyteCatalog catalog,
+                                   final AirbyteFileOffsetBackingStore offsetManager,
+                                   final Optional<AirbyteSchemaHistoryStorage> schemaHistoryManager) {
+    this.properties = properties;
+    this.config = config;
+    this.catalog = catalog;
+    this.offsetManager = offsetManager;
+    this.schemaHistoryManager = schemaHistoryManager;
+  }
+
+  protected Properties getDebeziumProperties() {
+    final Properties props = new Properties();
+    props.putAll(properties);
+
+    // debezium engine configuration
+    props.setProperty("name", "engine");
+    props.setProperty("offset.storage", "org.apache.kafka.connect.storage.FileOffsetBackingStore");
+    props.setProperty("offset.storage.file.filename", offsetManager.getOffsetFilePath().toString());
+    props.setProperty("offset.flush.interval.ms", "1000"); // todo: make this longer
```

Comment:
should this be configurable? / what's preventing it from being longer right now?

Reply:
I wouldnt want users to configure this. Its basically how often we want debezium to flush the state in the file. Debezium may or may not flush it though. It will try to but its based on other factors as well. The lower the value the lower the performance in wal processing. We never got back to finalising the right value. Also it would be out of scope for this PR